saleor/static/dashboard-next/components/TextFieldWithChoice/TextFieldWithChoice.tsx

```diff
@@ -43,89 +43,89 @@ const TextFieldWithChoice = withStyles(styles, {
-                            {...TransitionProps}
-                            style={{
-                              transformOrigin:
-                                placement === "bottom"
-                                  ? "right top"
-                                  : "right bottom"
-                            }}
-                          >
-                            <Paper>
-                              <ClickAwayListener
-                                onClickAway={closeMenu}
-                                mouseEvent="onClick"
-                              >
-                                <Menu>
-                                  {ChoiceProps.values.map(choice => (
-                                    <MenuItem
-                                      onClick={() => handleSelect(choice.value)}
-                                    >
-                                      {choice.label}
-                                    </MenuItem>
-                                  ))}
-                                </Menu>
-                              </ClickAwayListener>
-                            </Paper>
-                          </Grow>
-                        )}
-                      </Popper>
-                    </>
-                  );
-                }}
-              </MenuToggle>
-            )
-          }}
-        />
-      )}
-    </Anchor>
-  )
+                return (
+                  <>
+                    <div
+                      className={classes.adornment}
+                      ref={anchor}
+                      onClick={!menuOpen ? openMenu : undefined}
+                    >
+                      <Typography component="span" variant="caption">
+                        {ChoiceProps.label}
+                      </Typography>
+                      <DropdownIcon />
+                    </div>
+                    <Popper
+                      open={menuOpen}
+                      anchorEl={anchor.current}
+                      transition
+                      disablePortal
+                      placement="bottom-end"
+                      className={classes.menu}
+                    >
+                      {({ TransitionProps, placement }) => (
+                        <Grow
+                          {...TransitionProps}
+                          style={{
+                            transformOrigin:
+                              placement === "bottom"
+                                ? "right top"
+                                : "right bottom"
+                          }}
+                        >
+                          <Paper>
+                            <ClickAwayListener
+                              onClickAway={closeMenu}
+                              mouseEvent="onClick"
+                            >
+                              <Menu>
+                                {ChoiceProps.values.map(choice => (
+                                  <MenuItem
```

Comment:
Don't U have errors for missing `key` prop?

Reply:
Didn't notice any, but yeah it should have key 🙃